### PR TITLE
feat: added option to set neovim as default editor

### PR DESCRIPTION
**Pull request by itsOnly1Jah**
Noticed that the `defaultEditor` option was missing.

### CODE REVIEW

**Comment by shekohex** [comment]
Hey, is there any updates on this? @itsOnly1Jah @jooooscha

**Comment by itsOnly1Jah** [comment]
> Hey, is there any updates on this? @itsOnly1Jah @jooooscha 

It works, just waiting for it to merged into main branch.

**Comment by jooooscha** [comment]
I have added a review, can you see that?

**Comment by itsOnly1Jah** [comment]
> I have added a review, can you see that?

No, I can not see any review

File: nixneovim.nix [hunk 1]
```diff
@@ -151,6 +157,8 @@ in
         plugins = cfg.extraPlugins;
       };
 
+      environment.variables.EDITOR = mkIf cfg.defaultEditor (mkOverride 900 "nvim");
```
Comment (jooooscha) [inline nixneovim.nix:160]: Why is this needed? Isn't this handled by the `programs.neovim.defaultEditor` config?

**Comment by jooooscha** [comment]
Sorry, my mistake. You should be able to see the review now.

Apparently, GitHub does not include the author of the PR into the review process

File: nixneovim.nix (hunk 1 above)
Comment (itsOnly1Jah) [reply nixneovim.nix:160]: Its the same method used to set the `programs.neovim.defaultEditor`  but it eliminates redundancy since `nixneovim` already installs neovim.

Before:
```nix
    programs.neovim.defaultEditor = true;
    programs.nixneovim = {
       ...
     }
```
After:
```nix
    programs.nixneovim = {
        defaultEditor = true;
        ...
    }
```

Comment (jooooscha) [reply nixneovim.nix:160]: You set the `programs.neovim.defaultEditor` to `cfg.defaultEditor`.
However, `programs.neovim.defaultEditor` already sets the `EDITOR` variable` as seen [here](https://github.com/NixOS/nixpkgs/blob/5dab6490fe6d72b3f120ae8660181e20f396fbdf/nixos/modules/programs/neovim.nix#L166).

Therefore, It should not be necessary to set the `EDITOR` variable again, here. Right?

File: nixneovim.nix [hunk 2]
```diff
@@ -238,6 +246,7 @@ in
         {
           programs.neovim = {
             enable = true;
+            defaultEditor = cfg.defaultEditor;
```
Comment (jooooscha) [inline nixneovim.nix:249]: This only sets the default for home-manager.
Please also add this to the non-home-manager module

File: nixneovim.nix (hunk 1 above)
Comment (itsOnly1Jah) [reply nixneovim.nix]: Alright I understand what your saying, no need to recreate the variable since `programs.neovim.defaultEditor` already creates it.

File: nixneovim.nix (hunk 2 above)
Comment (itsOnly1Jah) [reply nixneovim.nix:247]: Added
